src/treemodel/nodelist.js

```diff
@@ -5,10 +5,67 @@
 
 'use strict';
 
-import Character from './character.js';
+import CharacterProxy from './characterproxy.js';
 import Text from './text.js';
-import Node from './node.js';
 import utils from '../utils.js';
+import langUtils from '../lib/lodash/lang.js';
+
+/**
+ * Value that is convertible to an item kept in {@link treeModel.NodeList} or an iterable collection of such items.
+ * In other words, this is anything that {@link treeModel.NodeList#constructor} is able to take and convert to node:
+ * * {@link treeModel.Element} will be left as is
+ * * {@link treeModel.CharacterProxy} will be left as is
+ * * {@link treeModel.Text} and {String} will be converted to a set of {@link treeModel.CharacterProxy}
+ * * {@link treeModel.NodeList} will clone a node list (but not the nodes inside, so the new and passed list will
+ * point to the same nodes.
+ * * Iterable collection of above items will be iterated over and all items will be added to the node list.
+ *
+ * @typedef {treeModel.Element|treeModel.CharacterProxy|treeModel.Text|String|treeModel.NodeList|Iterable} treeModel.NodesSet
+ */
+
+/**
+ * This is a private helper-class for {@link treeModel.NodeList} text compression utility.
+ *
+ * @private
```

Comment:
Because `characterProxy._nodeListText` is protected and used by the `TreeWalker` this class should be also `protected`, not `private`